src/libthread/thread.c

```diff
@@ -87,8 +87,7 @@ threadstart(uint y, uint x)
 	ulong z;
 
 //print("threadstart\n");
-	z = (ulong)x << 16;	/* hide undefined 32-bit shift from 32-bit compilers */
```

Comment:
@0intro I am not sure if I should have joined these two lines. Do you know their original intention? I assume that back on Plan9 `ulong`s were 32-bit? But I think that for plan9port they are always 64-bit?

Reply:
ulong are always 4-byte on Plan 9, but are either 4-byte (on 32-bit platforms) or 8-byte (on 64-bit platforms) where plan9port runs (Unix). I think this code was added to handle these two cases.

Reply:
@0intro I removed this particular change from the commit (it wasn't connected to the rest of the commit anyway), and added more fixes.